src/core/meshcop/secure_transport.cpp

```diff
@@ -159,18 +161,37 @@ Error SecureTransport::Open(ReceiveHandler aReceiveHandler, ConnectedHandler aCo
 
     mConnectedCallback.Set(aConnectedHandler, aContext);
     mReceiveCallback.Set(aReceiveHandler, aContext);
+    mShouldLimitConnectionAttempts = false;
+
     SetState(kStateOpen);
 
 exit:
     return error;
 }
 
+void SecureTransport::SetMaxConnectionAttempts(uint16_t aMaxAttempts, AutoCloseCallback aCallback, void *aContext)
+{
+    VerifyOrExit(mState != kStateClosed);
+
+    mShouldLimitConnectionAttempts = (aMaxAttempts != 0);
+    mRemainingConnectionAttempts   = aMaxAttempts;
+    mAutoCloseCallback.Set(aCallback, aContext);
+
+exit:
+    return;
+}
+
 Error SecureTransport::Connect(const Ip6::SockAddr &aSockAddr)
 {
     Error error;
 
     VerifyOrExit(IsStateOpen(), error = kErrorInvalidState);
 
+    if (mShouldLimitConnectionAttempts && (mRemainingConnectionAttempts > 0))
```

Comment:
nit: can we simplify this check?
```suggestion
    if (mRemainingConnectionAttempts > 0)
```